spec/Functions.tex

```diff
@@ -221,8 +221,7 @@ \section{Procedure Definitions}
 The \sntx{return-type} is optional and is discussed in~\rsec{Return_Types}.
 A type function may not specify a return type.
 
-The optional \sntx{where-clause} is only applicable if the function is
-generic.  It is discussed in~\rsec{Where_Expressions}.
+The optional \sntx{where-clause} is discussed in~\rsec{Where_Expressions}.
```

Comment:
I'd move it into Functions.tex (the "Procedures" chapter).  It looks to me like the current organization _roughly_ follows the order of the syntactic elements in declaring a procedure, which would suggest putting the where clause section after the "Return Types" section and before the "Nested Functions" section (but double-check me).